### PR TITLE
Update: add fixer for object-property-newline (fixes #7740)

### LINKED ISSUE #7740: Auto-fix support for object-property-newline
**What rule do you want to change?**

object-property-newline

**Does this change cause the rule to produce more or fewer warnings?**

It will produce the same number of warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

When the --fix flag is applied, code will be changed

**Please provide some example code that this change will affect:**

```js
const w = {x,
  y, z
}
```

As well as code without object shorthand.

```js
const w = {x: x,
  y: y, z: z
}
```

**What does the rule currently do for this code?**

Warns or fails depending on severity set.

**What will the rule do after it's changed?**

Take the above

```js
const w = {x,
  y, z
}
```
and transform it into
```js
const w = {
  x,
  y,
  z
}
```

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Add autofixing to a rule

**What changes did you make? (Give an overview)**

This adds an autofixer for [`object-property-newline`](http://eslint.org/docs/rules/object-property-newline), as described in https://github.com/eslint/eslint/issues/7740.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular

### CODE REVIEW

File: lib/rules/object-property-newline.js [hunk 1]
```diff
@@ -61,7 +63,17 @@ module.exports = {
                         context.report({
                             node,
                             loc: firstTokenOfCurrentProperty.loc.start,
-                            message: errorMessage
+                            message: errorMessage,
+                            fix(fixer) {
+                                const comma = sourceCode.getTokenBefore(firstTokenOfCurrentProperty);
+                                const rangeAfterComma = [comma.range[1], firstTokenOfCurrentProperty.range[0]];
+
+                                if (sourceCode.text.slice(rangeAfterComma[0], rangeAfterComma[1]).trim()) {
```
Comment (kaicataldo) [inline lib/rules/object-property-newline.js]: I think a comment explaining that the fixer bails when comments are present might be helpful to future contributors.

**Comment by vitorbal** [comment]
@platinumazure I think your concerns have been addressed now. Could you take a look when you have a moment?

File: tests/lib/rules/object-property-newline.js [hunk 2]
```diff
@@ -351,6 +374,30 @@ ruleTester.run("object-property-newline", rule, {
                     column: 13
                 }
             ]
+        },
+        {
+            code: "({ foo: 1 /* comment */, bar: 2 })",
+            output: "({ foo: 1 /* comment */,\nbar: 2 })", // not fixed due to comment
```
Comment (platinumazure) [inline tests/lib/rules/object-property-newline.js]: Looks like the comment here is inaccurate.

**Comment by gyandeeps** [comment]
@platinumazure If you are ok, then we can merge this.

**Comment by platinumazure** [comment]
Sorry for holding things up! :smile:

**Pull request merged**